docs/flow/reference/components/prometheus.relabel.md

```diff
@@ -0,0 +1,181 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/prometheus.relabel
+title: prometheus.relabel
+---
+
+# prometheus.relabel
+
+Prometheus metrics follow the [OpenMetrics](https://openmetrics.io/) format.
+Each time series is uniquely identified by its metric name, plus optional
+key-value pairs called labels. Each sample, i.e. a datapoint in the time
+series, contains a value and an optional timestamp.
+```
+<metric name>{<label_1>=<label_val_1>, <label_2>=<label_val_2> ...} <value> [timestamp]
+```
+
+The `prometheus.relabel` component rewrites the label set of each metric passed
+along to the exported receiver by applying one or more `metric_relabel_config`
+steps.  If no relabeling steps are defined or applicable to one of the metrics,
+then the metric will be forwarded as-is to each receiver passed in the
+component's arguments. If no labels are remaining after the relabeling steps
+are applied, then the metric will be dropped.
+
+The most common use of `prometheus.relabel` is to filter Prometheus metrics or
+standardize the label set that will be passed to one or more downstream
+receivers. The `metric_relabel_config` blocks will be applied to the label set
+of each metric in order of their appearance in the configuration file.
+
+Labels beginning with a double underscore are reserved for internal used and
+will be silently discarded after all relabeling steps have been applied.
+
+Multiple `prometheus.relabel` components can be specified by giving them
+different labels.
+
+## Example
+
+```river
+prometheus.relabel "keep_mysql_only" {
+  forward_to [prometheus.remote_write.onprem.receiver]
+
+  metric_relabel_config {
```

Comment:
Seeing this in action, `metric_relabel_config` feels a little awkward. I'm not sure what a better name for this would be. Maybe `rule`? 

Maybe it's not something we need to change for v0.28, but might be worth keeping in the back of our heads.

Reply:
We're talking about relabeling _rules_ and relabeling _steps_. I think that `rule` sounds reasonable, and maybe we should coordinate this with the discovery.relabel component, too.

We still have enough time, I believe it makes sense to change this before launch but I'd prefer to make bunch these changes in a different PR.

Reply:
> We still have enough time, I believe it makes sense to change this before launch but I'd prefer to make bunch these changes in a different PR.

👍

Reply:
I prefer `rule` over `metric_relabel_config` also but fine with shipping this without it.